proto/message_contents/public_key.proto

```diff
@@ -29,7 +30,8 @@ message UnsignedPublicKey {
 // SignedPublicKey 
 message SignedPublicKey {
     bytes key_bytes = 1;  // embeds an UnsignedPublicKey
-    Signature signature = 2; // signs key_bytes
+    Signature signature = 2; // signs key_bytes, legacy association proof
+    AssociationProof proof = 3; // proves association with a user identity
```

Comment:
Getting a bit out of my depth, but the more I think about it the associationProof seems like it is dependent on the keyBytes which hints that the abstraction layer might be a layer above.

If the goal is to move away from the generic term signatures, then replacing SignedPublicKey with AssociatedPublicKey could be beneficial. 

```
message AssociatedPublicKey {

    message V1 {
        bytes key_bytes
        bytes text
        enum SignatureType
        Signature
    }

    oneof union {
        v1 V1

```

From a roll out perspective both Bundles could be published in different topics until all clients can read the new AssociatedPublicKey format.

Pushback welcome

Reply:
It definitely makes sense theoretically, but I'm not sure what the tangible benefits of the `AssociatedPublicKey` concept are versus just adapting SignedPublicKey. I think managing multiple topics is a heavy lift.

Reply:
Fair.

>tangible benefits of the AssociatedPublicKey concept are versus just adapting SignedPublicKey
1) SignedPublicKey is not versioned which makes forward changes more difficult
2) Storing two fields makes it harder to deprecate old code path. How will the legacy Signature be removed in the future?

Reply:
> I think managing multiple topics is a heavy lift

FYI: The roll out procedure for larger changes in the past has been

If new types are needed:
- CreateNewTypes
- Have clients publish to a new topic (e.g. intro -> invite)
- Use the new types if conditions are met, else fallback to legacy
- deprecate legacy path once adoption is met.

Very open to new ideas here however this formulation has made type decoding easier as there is a one-to-one relationship between objects and topics. 

Ultimately having a SignedPublicKey.V2 is what I was striving for

Reply:
I don't see storing two fields making it any more difficult to deprecate the old code path. New clients don't need to check the old field at all, but will write both until eventually we draw a line in the sand on the old one. Should be identical to using a V2, just different code in different places. There is an argument to be made around which one is clearer to devs and I agree it could go either way.

I definitely think a SignedPublicKey.V2 would satisfy our goals, I just wonder if the scoping is right. The public key isn't changing at all (the curve, the key serialization format, the usage, etc). We're just changing the bytes that are used to verify the key is associated with an identity.